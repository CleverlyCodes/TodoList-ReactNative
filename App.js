import React, { useEffect, useState } from 'react';
import {
  Alert,
} from 'react-native';

import Auth from './screens/AuthScreen';
import Task from './screens/TaskScreen';

import * as LocalAuthentication from 'expo-local-authentication';

export default function App() {
  const [isAuthenticated, setIsAuthenticated] = useState(false);

  useEffect(() => {
    (async () => {
      const compatible = await LocalAuthentication.hasHardwareAsync();
      setIsBiometricSupported(compatible);
    })();
  });

  function onAuthenticate () {
    const auth = LocalAuthentication.authenticateAsync({
      promptMessage: "Authenticate first...",
      fallbackLabel: "Enter Password",
    });

    auth.then(result => {
      setIsAuthenticated(result.success);

      if (result.error) {
        Alert.alert("Login Failed!");
      } 
    });
  }

  return (
    <>
      {
        isAuthenticated ? <Task /> : <Auth onAuthenticate={onAuthenticate} />
      }
    </>
  );
}
