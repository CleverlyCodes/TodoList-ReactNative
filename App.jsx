import React, { useEffect, useState } from 'react';
import {
  Alert,
} from 'react-native';

import { setIsBiometricSupported, hasHardwareAsync, authenticateAsync } from 'expo-local-authentication';

import Auth from './screens/AuthScreen';
import Task from './screens/TaskScreen';

export default function App() {
  const [isAuthenticated, setIsAuthenticated] = useState(false);

  useEffect(() => {
    (async () => {
      const compatible = await hasHardwareAsync();
      setIsBiometricSupported(compatible);
    })();
  });

  function onAuthenticate() {
    const auth = authenticateAsync({
      promptMessage: 'Authenticate first...',
      fallbackLabel: 'Enter Password',
    });

    auth.then((result) => {
      setIsAuthenticated(result.success);

      if (result.error) {
        Alert.alert('Login Failed!');
      }
    });
  }

  return (
    { 
      isAuthenticated ? <Task /> : <Auth onAuthenticate={onAuthenticate} /> 
    }
  );
}
