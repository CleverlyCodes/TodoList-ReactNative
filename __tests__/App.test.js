import React from "react";
import renderer from "react-test-renderer";
import App from "../App.jsx"

describe("<App />", () => {
    it('Renders the root of the application', () => {
      const tree = renderer.create(<App />).toJSON();

      expect(tree).toMatchSnapshot();
    });
});