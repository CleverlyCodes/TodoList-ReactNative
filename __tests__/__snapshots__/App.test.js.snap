// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> has 2 child 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text>
    Please authenticate first
  </Text>
  <View
    accessibilityRole="button"
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Login
      </Text>
    </View>
  </View>
</View>
`;
